src/user-guide/writing-workflows/jinja2.rst

```diff
@@ -7,34 +7,28 @@ Jinja2
 .. tutorial::
    Configuration Consolidation Tutorial <tutorial-cylc-consolidating-configuration>
 
-Cylc supports use of the `Jinja2`_ template processor in workflow
-configurations. Jinja2 variables, expressions, loop control structures,
-conditional logic, etc., are processed to generate the final configuration. To
-the template processor, Jinja2 code is embedded in arbitrary text, but the
-result after processing must be valid Cylc syntax.
+Cylc supports the `Jinja2`_ template processor in workflow configurations.
+Jinja2 code can appear anywhere in the file. The result after Jinja2 processing
+must be valid Cylc syntax.
 
 To use Jinja2, put a hash-bang comment in the first line of :cylc:conf:`flow.cylc`:
 
 .. code-block:: cylc
 
    #!jinja2
 
-Template processing is the first thing done on parsing a workflow configuration
-so Jinja2 can appear anywhere in the file.
-
 Embedded Jinja2 code should be reasonably easy to understand for those with
-coding experience; but if not, Jinja2 is well documented `here
-<Jinja2_>`_.
+coding experience; otherwise Jinja2 is documented `here <Jinja2_>`_.
 
 Uses of Jinja2 in Cylc include:
 
  - Inclusion or exclusion of config sections by logical switch, e.g. to make
```

Comment:
There are other entirely different reasons for doing this (rose stem suites use this as a mechanism for selecting tests from a test-battery) and none of the others have examples. This is also an example for bullet point 3. Can we cut it?

Reply:
> There are other entirely different reasons for doing this 

Well, it says "e.g.", which suggests there are other uses.  We could extend this section to explicitly mention other use cases, but that's a bit off topic for this PR (admittedly the PR title sounds more general than it should!).

> This is also an example for bullet point 3. Can we cut it?

Do you mean "Inclusion of files and sub-templates"?  Bullet point one is about switching on/off inlined config sections.

Reply:
> Do you mean "Inclusion of files and sub-templates"? Bullet point one is about switching on/off inlined config sections.

Either method could be used for site-switching. 


NVM, just random thoughts rather than things wrong with this PR.